### PR TITLE
adding pmocz as developer to docs / doc typos

### PR DESCRIPTION
**Pull request by pmocz**
Adding Philip Mocz (me) to the docs as a new developer. I caught a few typos reading the docs so I am fixing them too

### CODE REVIEW

**Review by Debraheem — APPROVED** [review]
looks good!

File: docs/source/about.rst
Comment (evbauer) [inline docs/source/about.rst:1]: Heads up that per policy, there's some more process that needs to be followed before we can formally add a new "MESA Developers" member to this part of the documentation: https://docs.mesastar.org/en/release-r24.03.1/developing/new_developers.html

**Review by orlox — COMMENTED** [review]
Looks good for me, except for the comment from @evbauer

File: docs/source/about.rst
Comment (pmocz) [reply docs/source/about.rst:1]: Thanks for pointing this out, Evan! I will merge this PR once all those steps are complete

**Review by mjoyceGR — APPROVED** [review]

**Pull request merged**